peers/app_request_network.go

```diff
@@ -225,43 +220,39 @@ func (n *AppRequestNetwork) ConnectPeers(nodeIDs set.Set[ids.NodeID]) (set.Set[i
 					zap.String("beaconIP", peer.PublicIP),
 					zap.Error(err),
 				)
-				retErr = fmt.Errorf("failed to connect to peers: %v", err)
 				continue
 			}
 			trackedNodes.Add(peer.ID)
 			n.Network.ManuallyTrack(peer.ID, ipPort)
 			if len(trackedNodes) == nodeIDs.Len() {
-				return trackedNodes, retErr
+				return trackedNodes
 			}
 		}
 	}
 
-	// attempt adding api node in case it is a validator
+	// If we haven't yet connected to all peers, attempt to connect to the API node in case it is a validator
```

Comment:
If the API node is a validator, wouldn't it's node ID already have been included in `nodeIDs`? I find it a little odd that the caller here gets to specify a set of node IDs to connect to but the implementation may add to that set.

Reply:
If we reach this point, the only thing we'd want to do is manually track the Info API node as a potential validator. The passed in list of `nodeIDs` is the full set of subnet validators as fetched from the P-Chain, so we should definitely not mutate that list. I think what's missing here are validation checks that the fetched Info API node ID is in fact in the validator set. I'll add that check in.

Reply:
@michaelkaplan13 let me know if this approach makes sense to you. We're not mutating the input list at all.

Reply:
Rewrote the comment to make it clear that we're checking for the case in which the API node is in `nodeIDs`

Reply:
Ah, yeah that makes sense, thanks.